flaml/nlp/autotransformers.py

```diff
@@ -0,0 +1,787 @@
+                resplit_portion:
+                    the proportion for resplitting the train and dev data when split_mode="resplit". Not required.
+            '''
+        self._max_seq_length = max_seq_length
+        self._server_name = server_name
+        self.jobid_config = jobid_config
+        self.wandb_utils = wandb_utils
+
+        self.path_utils = PathUtils(jobid_config, hpo_data_root_path = data_root_path)
+
+        if jobid_config.spt == "rspt":
+            assert resplit_portion, "If split mode is 'rspt', the resplit_portion must be provided. Please " \
+                                    "refer to the example in the documentation of AutoTransformers.prepare_data()"
+        if jobid_config.subdat:
+            data_raw = load_dataset(jobid_config.dat[0], jobid_config.subdat)
+        else:
+            data_raw = self._wrapper(load_dataset, *jobid_config.dat)
+
+        self._train_name, self._dev_name, self._test_name = self._get_split_name(data_raw, fold_name=fold_name)
+        auto_tokentoids_config = {"max_seq_length": self._max_seq_length}
+        self._tokenizer = AutoTokenizer.from_pretrained(jobid_config.pre_full, use_fast=True)
+
+        data_encoded = AutoToEncoded.from_model_and_dataset_name(data_raw,jobid_config.pre_full,jobid_config.dat[0], jobid_config.subdat, **auto_tokentoids_config)
+        self._max_seq_length = 0
+        for each_fold in data_encoded.keys():
+            self._max_seq_length = max(self._max_seq_length,
+                max([sum(data_encoded[each_fold][x]['attention_mask']) for x in range(len(data_encoded[each_fold]))]))
+        self._max_seq_length = int((self._max_seq_length + 15) / 16) * 16
+        data_encoded = AutoToEncoded.from_model_and_dataset_name(data_raw, jobid_config.pre_full, jobid_config.dat[0], jobid_config.subdat,**auto_tokentoids_config)
+
+        if jobid_config.spt == "rspt":
+            all_folds_from_source = []
+            assert "source" in resplit_portion.keys(), "Must specify the source for resplitting the dataset in" \
+            "resplit_portion, which is a list of folder names, e.g., resplit_portion = {'source': ['train']}"
+
+            source_fold_names = resplit_portion['source']
+            for each_fold_name in source_fold_names:
+                this_fold_dataset = data_encoded[each_fold_name]
+                all_folds_from_source.append(this_fold_dataset)
+
+            merged_folds_from_source = datasets.concatenate_datasets(all_folds_from_source)
+            merged_folds_from_source = merged_folds_from_source.shuffle(seed=42)
+
+            assert "train" in resplit_portion.keys() and "validation" in resplit_portion.keys() \
+                and "test" in resplit_portion.keys(), "train, validation, test must exist in resplit_portion"
+
+            for key in ["train", "validation", "test"]:
+                target_fold_start, target_fold_end = int(resplit_portion[key][0] * len(merged_folds_from_source)), \
```

Comment:
having two numbers in the portion for each key is unnecessary

Reply:
minor